server/rabbitmq.go

```diff
@@ -140,20 +142,30 @@ func (s *AMQPServer) handleRequest(delivery amqp.Delivery) {
 		return
 	}
 
-	body, err := proto.Marshal(resp)
+	// Marshal the response
+	// @todo we're currently always marshaling errors as proto
```

Comment:
In the short term we can address that on the API level. There already is a method to take an error and render it in json